### DISCUSSION TITLE
ale_linters use g++ for header files lead to error

### DISCUSSION THREAD

**Discussion by laoshaw** [w0rp/ale#4330] [General]
I use g++ as the linter for my c++ code which worked well for c++ code, but for header files it will report 'g++-11: fatal error: cannot use ‘-’ as input filename for a precompiled header`

looking at ALEInfo, I saw it is doing: `g++-11 -S -x c++-header  - < test.hpp` which caused above error. However I don't recall I saw this problem in the past with c++ headers.

is this a new problem with the newest ale? or I have been wrong all along. is there a fix?

**Comment by hsanson** [comment]
Not entirely sure but this was recently merged to ALE that seems related: https://github.com/dense-analysis/ale/pull/4318
